examples/text-classification/run_xnli.py

```diff
@@ -303,44 +388,50 @@ def evaluate(args, model, tokenizer, prefix=""):
     return results
 
 
-def load_and_cache_examples(args, task, tokenizer, evaluate=False):
-    if args.local_rank not in [-1, 0] and not evaluate:
+def load_and_cache_examples(model_args, data_args, training_args, task, tokenizer, evaluate=False):
```

Comment:
Thank you for working on this!

But it seems that the dataset is still loaded using the old way without using the `datasets` library.

Also IMO it would be better, if we update the script to use `Trainer`, like `run_glue.py`, let us know if you want to work on this, happy to help here :)

Reply:
I see, use the Trainer() let me see if I could figure it out. thanks for the tip @patil-suraj

Reply:
@patil-suraj I loaded dataset at line 501 in `main()` using `load_dataset(data_args.task_name, model_args.language, data_dir=data_args.data_dir)` but only used `data_dir` here to strictly to cache `cached_feature_file`. Am I still downloading the dataset `xnli` incorrectly?

Reply:
Hi @patil-suraj  I took another crack at the PR. I tried to adapt `run_glue.py` as closely to the task of `fine-tuning mBERT (multi-lingual BERT) on the XNLI dataset` as best as I could. As these options below weren't in the original run_xnli.py, I did not include them:
- option to use csv or json train/test files from user
- do_predict   

I also assumed that this script is solely to run xnli task. Hopefully it's closer to what you required :)
Let me know if there is anything you wish to change.